### PR TITLE
Fix build status image update issue

### PR DESCRIPTION
**Pull request by sorta**
For https://travisci.assembla.com/spaces/VCS/tickets/145/details

I've been testing like this:
1. Create a new repo on provider with simple tests. A basic ember project works well, and to make this faster I've just been forking one that I already have.
2. Activate repo on travis if needed
3. Go to the repo page and trigger a build. Watch as the build completes, observing the status button.
4. Flip through tabs on completion

This regularly reproduces the issue on prod for me. I suspect it may be a caching thing, so I added a status-based query param to the image displayed on `travis-web`. Compare:
https://travis-ci.com/
https://so-fix-build-status-img-2.test-deployments.travis-ci.com/

I've seen this for both github and bitbucket, but if you want to test with bb, I would test on staging, which i did deploy to recently
https://staging.travis-ci.com/
For some reason, bitbucket status images seem generally broken on test deployments, the API is returning 403 for them for some reason. This doesn't seem to happen on prod and stag though.

This is an alternative to this PR https://github.com/travis-ci/travis-web/pull/2414 which I will probably close in favor of this one. I realized that my solution on 2414 would change the URL being given to users, not just the URL being used to display the badge on our site, which I don't think we necessarily want.

### CODE REVIEW

File: app/components/repo-status-badge.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  statusImageUrl: computed(
+    'repo.slug',
+    'repo.private',
+    'defaultBranch',
+    'lastBuildState',
+    function () {
+      const { defaultBranch, lastBuildState = 'unknown', repo } = this;
+
+      const url = this.statusImages.imageUrl(repo, defaultBranch);
+      const divider = (url.indexOf('?') >= 0) ? '&' : '?';
```
Comment (artursmirnov) [inline app/components/repo-status-badge.js]: I think it's pretty safe to use
```suggestion
      const divider = url.includes('?') ? '&' : '?';
```
What do you think?

File: app/templates/components/repo-status-badge.hbs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<div class="repo-badge inline-block vertical-align">
+  <a
+    title="Latest push build on default branch: {{this.lastBuildState}}"
+    name="status-images"
+    class="pointer open-popup"
+    data-test-status-image-popup="true"
+    {{on 'click' this.onClick}}
```
Comment (artursmirnov) [inline app/templates/components/repo-status-badge.hbs]: As long as it's an external attribute, what do you think about
```suggestion
    {{on 'click' @onClick}}
```
?

File: app/components/repo-status-badge.js (hunk 1 above)
Comment (sorta) [reply app/components/repo-status-badge.js]: Ahh yeah, that sounds good to me :)

File: app/templates/components/repo-status-badge.hbs (hunk 2 above)
Comment (sorta) [reply app/templates/components/repo-status-badge.hbs]: Totally, thanks for catching it!

**Comment by sorta** [comment]
Thanks for the reviews everyone!

**Pull request merged**